components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/TokenValidationHandler.java

```diff
@@ -458,7 +460,23 @@ private OAuth2IntrospectionResponseDTO validateAccessToken(OAuth2TokenValidation
                         = OAuthServerConfiguration.getInstance().isCrossTenantTokenIntrospectionAllowed();
                 if (!isCrossTenantTokenIntrospectionAllowed && accessTokenDO != null &&
```

Comment:
Is enabling cross-tenant introspection necessary for the organization management feature? While cross-tenant introspection is often used for SaaS applications, it may not be relevant for B2B shared applications.WDYT?

Reply:
`Is enabling cross-tenant introspection necessary for the organization management feature` no
Here we are B2B application token handling under isCrossTenantTokenIntrospectionAllowed = false condition

Reply:
Sorry I missed it but my point is B2B token introspection is orthogonal to `isCrossTenantTokenIntrospectionAllowed` & it should not depend on the value of the `isCrossTenantTokenIntrospectionAllowed` at all, may if we can do the `isOrganizationManagementEnabled` logic first and if it is a success then use another variable skip the error throwing part from `isCrossTenantTokenIntrospectionAllowed` logic